include/openspace/scene/scenegraphnode.h

```diff
@@ -63,21 +65,32 @@ class SceneGraphNode : public properties::PropertyOwner {
     SceneGraphNode();
     ~SceneGraphNode();
 
-    static SceneGraphNode* createFromDictionary(const ghoul::Dictionary& dictionary);
+    static std::unique_ptr<SceneGraphNode> createFromDictionary(const ghoul::Dictionary& dictionary);
 
     bool initialize();
     bool deinitialize();
 
+    void traversePreOrder(std::function<void(SceneGraphNode*)> fn);
```

Comment:
These are standard names for traversal orders: https://en.wikipedia.org/wiki/Tree_traversal#Pre-order_2